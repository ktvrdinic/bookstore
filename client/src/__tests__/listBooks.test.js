import React from 'react';
import ListBooks from '../components/globals/listBooks';
import { createShallow } from '@material-ui/core/test-utils';

describe('Test List book component', () => {
    let shallow, wrapper;

    beforeAll(() => {
        shallow = createShallow();
    });

    it('Should not render any cards', () => {
        wrapper = shallow(<ListBooks books={[]} />);
        expect(wrapper.find('#card')).toHaveLength(0);
    });

    it('Should render right number of cards', () => {
        let books = [{
            title: 'Test',
            description: 'This is interesting book',
            coverImg: '',
            price: 222,
            _id: 1
        },
        {
            title: 'Test2',
            description: 'This is interesting book2',
            coverImg: '',
            price: 333,
            _id: 2
        }];

        wrapper = shallow(<ListBooks books={books} />);
        expect(wrapper.find('#card')).toHaveLength(2);
    });

    it('Should pass and place text correctly', () => {
        let books = [{
            title: 'Test',
            description: 'This is interesting book',
            coverImg: '',
            price: 222,
            _id: 1
        }];

        wrapper = shallow(<ListBooks books={books} />);
        expect(wrapper.find('#title').text()).toEqual(books[0].title);
        expect(wrapper.find('#description').text()).toEqual(books[0].description);
        expect(wrapper.find('#cardMedia').text()).toEqual(books[0].coverImg);
    });
})
