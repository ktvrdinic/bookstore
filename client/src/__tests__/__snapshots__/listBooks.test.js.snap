// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Entry point Should have a search tag with Search field! 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
